From 9d08a6683952f6b9fcbccc1f079783d6823137ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bernhard=20Rosenkr=C3=A4nzer?= <bero@lindev.ch>
Date: Fri, 11 May 2018 00:22:43 +0200
Subject: [PATCH] Fix build with android-p-preview-2 libsparse

Fix sparse_import_segment function signature to match
android-p-preview-2's libsparse

Signed-off-by: Bernhard Rosenkraenzer <bero@lindev.ch>
---
 lib/libf2fs_io.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/libf2fs_io.c b/lib/libf2fs_io.c
index 4781517..1ed5ca5 100644
--- a/lib/libf2fs_io.c
+++ b/lib/libf2fs_io.c
@@ -114,7 +114,7 @@ static int sparse_write_blk(__u64 block, int count, const void *buf)
 	return 0;
 }
 
-static int sparse_import_segment(void *UNUSED(priv), const void *data, int len,
+static int sparse_import_segment(void *UNUSED(priv), const void *data, size_t len,
 		unsigned int block, unsigned int nr_blocks)
 {
 	/* Ignore chunk headers, only write the data */
-- 
2.17.0

